### PR TITLE
Set GAPTypes and GapObj global variables in init, before loading GAP

### PR DESCRIPTION
**Pull request by sebasguts**
Fixes [this problem](https://github.com/homalg-project/HomalgProject.jl/issues/2)

### CODE REVIEW

File: src/GAP.jl [hunk 1]
```diff
@@ -111,6 +112,8 @@ function __init__()
     ## the module is loaded.
     gap_module = @__MODULE__
     Base.MainInclude.eval(:(__JULIAGAPMODULE = $gap_module))
+    Base.MainInclude.eval(:(__GapObj_TYPE_VAR = $(gap_module.GAPTypes.GapObj)))
+    Base.MainInclude.eval(:(__GapObj_MODULE_VAR = $(gap_module.GAPTypes)))
```
Comment (fingolfin) [inline src/GAP.jl]: Could you please elaborate what's going on here? Why do these mysterious globals __GapObj_TYPE_VAR and __GapObj_MODULE_VAR have to be set (nothing else references them?)

Comment (sebasguts) [reply src/GAP.jl]: Sorry for the missing link. The variables are used [here](https://github.com/gap-system/gap/pull/3669)

Comment (fingolfin) [reply src/GAP.jl]: But then why is this PR here needed at all? We already have `__JULIAGAPMODULE` aka `gap_module ` on the GAP side, it should be no problem to extract its members / submodules `gap_module.GAPTypes` and `gap_module.GAPTypes.GapObj` from that, no?

**Comment by sebasguts** [comment]
The corresponding GAP pr does not need these changes any more, therefore they can be ignored.

**Pull request closed without merging**